alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -219,41 +233,39 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
                     // Prepare DocumentExecutionContext for each index
                     val docExecutionContext = DocumentExecutionContext(queries, indexLastRunContext, indexUpdatedRunContext)
 
-                    val matchingDocs = getMatchingDocs(
+                    fetchShardDataAndMaybeExecutePercolateQueries(
```

Comment:
This method has a lot of parameters. Consider using an object as the parameter. That way a builder could be used so that the order of arguments does not matter. Would also make this more readable.